integration/experiment/ffnet/README.md

```diff
@@ -0,0 +1,243 @@
+
+#### `gen_neural_net.py`:
+
+   Generates neural net json file(s) for CPU and/or GPU. This takes in the trace h5 file generated 
+   from `gen_hdf_from_fsweep.py` which is annotated with region classes. Required signals are 
+   specified within this script, per-domain. The script checks for the complete list of signals
+   and generates the corresponding neural nets.
+
+   This script can take in the following inputs:
+
+   - `output`: Prefix of the output json file(s).
+   - `description`: Description of the neural net
+   - `ignore` : A comma-separated list of region hashes to ignore
+   - `data` : Data files to train on. This can take in multiple trace HDF files.
+
+   Note that this script depends upon pytorch. Pytorch can be installed using: `pip install pytorch`
+  
+   E.g.:
+
+   ```
+   ./gen_neural_net.py --output nnet --description "A neural net" --data example_output_traces.h5
+   ```
+   This will generate `nnet_cpu.json` and/or `nnet_gpu.json`
+
+#### `gen_region_parameters.py`:
+
+   Generates region-class frequency recommendation map. This takes in the stats h5 file
+   generated from `gen_hdf_from_fsweep.py`, which is annotated with region classes and
+   contains runtime and per-domain energy information. This is used to generate a
+   runtime-frequency relationship and determine minimum energy point within an allowable
+   perf degradation. For each region class, a list is generated which contains the 
+   desired frequency for a region class for various perf-energy-bias values, from
+   most perf-sensitive to most energy-sensitive.
+
+   This script takes in the following input:
+
+   - `output`: Prefix of the output json file(s).
+   - `data-file`: The stats HDF generated from `gen_hdf_from_fsweep.py`.
+   
+   E.g.:
+
+   ```
+    ./gen_region_parameters.py --output fmap --data-file example_output_stats.h5
+   ```
+
+   This will generate `fmap_cpu.json` and/or `fmap_gpu.json`
+
+#### `gen_sweep_to_ffnet.py`:
+
+   Executes scripts referenced above to: Generate HDFs, generate neural nets, and generate
+   region-class frequency recommendation maps.
+
+   This script takes in the following input:
+
+   - `output`: prefix of the output HDF and json file(s)
+   - `description`: Description of the neural net
+   - `ignore` : A comma-separated list of region hashes to ignore
+   - `frequency_sweep_dirs`: Directories containing reports and traces from frequency sweeps
+
+   This will generate all HDFs and json files referenced above.
+  
+   E.g. 
+   ```
+   ./gen_sweep_to_ffnet.py --output test --description "Test" --frequency_sweep_dirs /path/to/fsweep
+   ```
+   This will generate the following files:
+   - `test_stats.h5`
+   - `test_traces.h5`
+   - `test_nn_cpu.json`  and/or `test_nn_gpu.json`
+   - `test_fmap_gpu.json` and/or `test_fmap_gpu.json`
+   
+
+## Scripts to Produce Neural Nets and Frequency Recommendation Maps
+
+The end-to-end process for utilizing this agent to improve energy efficiency can be conducted
```

Comment:
This is the README for the ffnet experiment, rather than for the agent

```suggestion
The end-to-end process for utilizing this experiment to improve energy efficiency can be conducted
```

or

```suggestion
The end-to-end process for utilizing the `ffnet` agent to improve energy efficiency can be conducted
```